src/page/ConfigPageLayout/tabs/SecretsManagementTab/SecretsManagementUI.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import { Button, ConfirmModal, EmptyState } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import { SecretWithUuid } from './types';
+import { useDeleteSecret, useSecrets } from 'data/useSecrets';
+import { CenteredSpinner } from 'components/CenteredSpinner';
+
+import { ConfigContent } from '../../ConfigContent';
+import { SECRETS_EDIT_MODE_ADD } from './constants';
+import { SecretCard } from './SecretCard';
+import { SecretEditModal } from './SecretEditModal';
+
+export function SecretsManagementUI() {
+  const [editMode, setEditMode] = useState<string | false>(false);
+  const [deleteMode, setDeleteMode] = useState<SecretWithUuid | undefined>();
+  const { data: secrets, isLoading, isFetching } = useSecrets();
+  const deleteSecret = useDeleteSecret();
+  const emptyState = secrets?.length === 0;
+
+  const handleAddSecret = () => {
+    setEditMode(SECRETS_EDIT_MODE_ADD);
+  };
+
+  const handleEditSecret = (id?: string) => {
+    setEditMode(id ?? false);
+  };
+
+  const handleDeleteSecret = (id: string) => {
+    const secret = secrets?.find((s) => s.uuid === id);
+    if (secret) {
+      setDeleteMode(secret);
+    }
+  };
+
+  if (isLoading) {
+    return <ConfigContent loading ariaLoadingLabel="Loading secrets" />;
+  }
+
+  return (
+    <>
+      {emptyState ? (
+        <ConfigContent>
+          <EmptyState
+            variant="call-to-action"
+            message="You don't have any secrets yet."
+            button={
+              <Button onClick={handleAddSecret} icon="plus">
+                Create secret
+              </Button>
+            }
+          >
+            You can use secrets to store sensitive information such as passwords, API keys, and other sensitive
+            information.
```

Comment:
```suggestion
            You can use secrets to store private information such as passwords, API keys, and other sensitive
            data.
```
Just to avoid some of the repetition and make the copy read a little better.